docs/src/main/sphinx/connector/non-transactional-merge.fragment

```diff
@@ -0,0 +1,10 @@
+### Non-transactional MERGE
+
+The connector supports adding rows using [MERGE statements](/sql/merge), if the
+`merge.non-transactional-merge.enabled` catalog property or the corresponding
+`non_transactional_merge_enabled` catalog session property is set to 
+`true`. Merge is only supported for directly modifying target tables.
```

Comment:
Not sure this sentence makes sense and takes the original content and captures what was meant @chenjian2664 and @ebyhr